Author: Martin Koßler <martin.kossler@students.uni-marburg.de>
This patch only works with context-menu patch applied. 
It includes new command line arguments and about-data
--- main.cpp.orig	2007-02-16 20:41:04.000000000 +0100
+++ main.cpp	2007-03-15 14:04:53.000000000 +0100
@@ -20,20 +20,50 @@
 
 #include <qapplication.h>
 #include <kapplication.h>
+#include <kaboutdata.h>
+#include <kcmdlineargs.h>
+#include <kaboutapplication.h>
+#include <qimage.h>
 #include <vector>
-//#include "amfs.h" 
 #include "amfs-xml.h"
+#include "amarokfs_config.h"
 
 using namespace std;
 
 int main( int argc, char ** argv )
 {
-    KApplication a( argc, argv, "amarokFS" );
-	//Amfs *x = new Amfs();
-	AmfsXML *x = new AmfsXML();
+	KAboutData* about = new KAboutData("amarokFS", "Amarok Full Screen", "0.4.2",
+		"A full screen front end for Amarok", KAboutData::License_GPL_V2,
+		"© 2007, Jan Zarnikov", 0,
+		"http://www.kde-apps.org/content/show.php?content=52641");
+	about->addAuthor("Jan Zarnikov", 0, "jan.zarnikov@gmail.com");
+	about->addAuthor("Martin Koßler", 0, "martin.kossler@students.uni-marburg.de");
+	KCmdLineArgs::init(argc, argv, about);
+	KCmdLineOptions options[] =
+	{
+	    { "settings", "show config dialog", 0 },
+	    { "kiosk", "disable Settings entry in contextmenu", 0 },
+	    KCmdLineLastOption
+	};
+	KCmdLineArgs::addCmdLineOptions(options);
+	KCmdLineArgs* args = KCmdLineArgs::parsedArgs();
+
+	KApplication a;
+	QWidget* x;
+
+	if(args->isSet("settings")) {
+		x = new AmarokFS_config();
+		x->show();
+	} else {
+		x = new AmfsXML();
+		if(args->isSet("kiosk"))
+			((AmfsXML*)x)->setupAct->setEnabled(FALSE);
+		((AmfsXML*)x)->aboutWindow = new KAboutApplication(about, x, "about", FALSE);
+		x->showFullScreen();
+	}
+
+	x->setIcon(QImage("/usr/share/icons/amarokFS.png"));
 	a.setMainWidget(x);
-	a.connect( &a, SIGNAL( lastWindowClosed() ), &a, SLOT( quit() ) );
-	x->showFullScreen();
 	return a.exec();
 }
 	
--- amfs-xml.h.orig	2007-03-15 15:58:36.000000000 +0100
+++ amfs-xml.h	2007-03-15 16:00:50.000000000 +0100
@@ -23,5 +23,6 @@
 #include <qaction.h>
 #include <qpopupmenu.h>
+#include <kaboutapplication.h>
 				
 #include <vector>
 #include <dcopclient.h>		
@@ -89,11 +90,13 @@
 	void updateTime();
 	void controlButtonClicked(QCString command);
 	void launchConfig();
+	void launchAbout();
 	void hideHint();
 	void hideMouse(bool now = TRUE);
 
 public:
 	QAction *reloadAct, *setupAct, *closeAct, *aboutAct;
+	KAboutApplication* aboutWindow;
 };
 
 #endif
--- amfs-xml.cpp.orig	2007-03-15 15:58:19.000000000 +0100
+++ amfs-xml.cpp	2007-03-15 16:00:44.000000000 +0100
@@ -4,6 +4,7 @@
 #include <kapp.h>
 #include <kconfig.h>
 #include <kiconloader.h>
+#include <qiconset.h>
 
 #include <qfile.h>
 #include <qdom.h>
@@ -115,6 +116,11 @@
 	setupAct->addTo(contextMenu);
 
 	contextMenu->insertSeparator();
+	aboutAct = new QAction(tr("&About"), 0, this);
+	aboutAct->setIconSet(QIconSet(QPixmap("/usr/share/icons/amarokFS.png")));
+	aboutAct->setStatusTip(tr("Show About Dialog."));
+	connect(aboutAct, SIGNAL(activated()), this, SLOT(launchAbout()));
+	aboutAct->addTo(contextMenu);
 	closeAct = new QAction(tr("&Quit"), tr("Esc"), this);
 	closeAct->setIconSet(SmallIconSet("exit"));
 	closeAct->setStatusTip(tr("Quit Amarok Full Screen."));
@@ -401,4 +407,9 @@
 	contextMenu->exec(event->globalPos());
 }
 
+void AmfsXML::launchAbout() {
+	if(aboutWindow != 0)
+		aboutWindow->show();	
+}
+
 void AmfsXML::launchConfig() {
